File: project/hyeok-onboarding/src/main/java/com/INNER_CIRCLE_ICD4/innerCircle/domain/QuestionType.java
package com.INNER_CIRCLE_ICD4.innerCircle.domain;

public enum QuestionType {
    SHORT,
    LONG,
    SINGLE_CHOICE,
    MULTIPLE_CHOICE;

    public boolean isChoiceType() {
        return this == SINGLE_CHOICE || this == MULTIPLE_CHOICE;
    }

    public boolean isTextType() {
        return this == SHORT || this == LONG;
    }
}

File: project/hajin-onboarding/src/main/kotlin/com/innercircle/survey/survey/domain/QuestionType.kt
package com.innercircle.survey.survey.domain

import com.innercircle.survey.survey.domain.exception.InvalidQuestionTypeException

enum class QuestionType(
    val description: String,
    val requiresChoices: Boolean,
) {
    SHORT_TEXT("단답형", false),

    LONG_TEXT("장문형", false),

    SINGLE_CHOICE("단일 선택", true),

    MULTIPLE_CHOICE("다중 선택", true),
    ;

    fun isTextType(): Boolean = !requiresChoices

    fun isChoiceType(): Boolean = requiresChoices

    companion object {
        fun fromString(value: String): QuestionType {
            return try {
                valueOf(value.uppercase())
            } catch (e: IllegalArgumentException) {
                throw InvalidQuestionTypeException(value)
            }
        }
    }
}
